frontend/src/component/admin/groups/GroupsList/GroupCard/GroupCardAvatars/NewGroupCardAvatars.tsx

```diff
@@ -43,101 +23,20 @@ interface IGroupCardAvatarsProps {
-    AvatarComponent,
-    ...props
+    header,
+    ...avatarGroupProps
 }: IGroupCardAvatarsProps) => {
-    const Avatar = StyledAvatar(AvatarComponent ?? UserAvatar);
-
-    return <GroupCardAvatarsInner AvatarComponent={Avatar} {...props} />;
-};
-
-type GroupCardAvatarsInnerProps = Omit<
-    IGroupCardAvatarsProps,
-    'AvatarComponent'
-> & {
-    AvatarComponent: typeof UserAvatar;
-};
-
-const GroupCardAvatarsInner = ({
-    users = [],
-    header = null,
-    avatarLimit = 9,
-    AvatarComponent,
-}: GroupCardAvatarsInnerProps) => {
-    const shownUsers = useMemo(
-        () =>
-            users
-                .sort((a, b) => {
-                    if (
-                        Object.hasOwn(a, 'joinedAt') &&
-                        Object.hasOwn(b, 'joinedAt')
-                    ) {
-                        return (
-                            (b as IGroupUser)?.joinedAt!.getTime() -
-                            (a as IGroupUser)?.joinedAt!.getTime()
-                        );
-                    }
-                    return 0;
-                })
-                .slice(0, avatarLimit),
-        [users],
-    );
-
-    const [anchorEl, setAnchorEl] = useState<HTMLElement | null>(null);
-    const [popupUser, setPopupUser] = useState<{
-        name: string;
-        description?: string;
-        imageUrl?: string;
-    }>();
-
-    const onPopoverOpen = (event: React.MouseEvent<HTMLElement>) => {
-        setAnchorEl(event.currentTarget);
-    };
-
-    const onPopoverClose = () => {
-        setAnchorEl(null);
-    };
-
-    const avatarOpen = Boolean(anchorEl);
-
     return (
         <StyledContainer>
             <ConditionallyRender
                 condition={typeof header === 'string'}
                 show={<StyledHeader>{header}</StyledHeader>}
                 elseShow={header}
             />
-            <StyledAvatars>
-                {shownUsers.map((user) => (
-                    <AvatarComponent
-                        key={objectId(user)}
-                        user={{ ...user, id: objectId(user) }}
-                        onMouseEnter={(event) => {
-                            onPopoverOpen(event);
-                            setPopupUser(user);
-                        }}
-                        onMouseLeave={onPopoverClose}
-                    />
-                ))}
-                <ConditionallyRender
-                    condition={users.length > avatarLimit}
-                    show={
-                        <AvatarComponent>
-                            +{users.length - shownUsers.length}
-                        </AvatarComponent>
-                    }
-                />
-                <GroupPopover
-                    open={avatarOpen}
-                    user={popupUser}
-                    anchorEl={anchorEl}
-                    onPopoverClose={onPopoverClose}
-                />
-            </StyledAvatars>
+            <AvatarGroup {...avatarGroupProps} />
```

Comment:
Feels like this should be tackled by composition, instead of prop inheritance 

```tsx
<GroupCardAvatars header="">
  <AvatarGroup ... />
</GroupCardAvatars>
```

Reply:
Right. So in that case, `GroupCardAvatars` doesn't actually provide any avatars; it's just a header? At that point, I don't think calling it `GroupCardAvatars` makes a lot of sense anymore. But maybe it doesn't and we should just delete this whole file?

Reply:
Yup. It's easier to reason about this when there is 1 less level of abstraction

Reply:
Less to reason about, but also more detailed knowledge required elsewhere. I'll have a look at what this'd look like later.